core/Container/ContainerFactory.php

```diff
@@ -82,4 +82,16 @@ private function addEnvironmentConfig(ContainerBuilder $builder)
 
         $builder->addDefinitions($file);
     }
+
+    private function addPluginConfigs(ContainerBuilder $builder)
+    {
+        $plugins = Config::getInstance()->Plugins['Plugins'];
```

Comment:
Loading config files (or even changing definitions) after the container has already been used is really risky. If any service or value has already been resolved and injected somewhere, it will not be replaced.

E.g. the logger is injected somewhere, then a plugin changes the logger: the old one is still injected in some places… Leading to inconsistent services.

It might not be a problem directly, and the logger might not be so critical, but it's calling for trouble later with more serious components or config values.

Ideally the container should be built in one step and entries should not be changed after that.